src/pip/_internal/utils/outdated.py

```diff
@@ -119,12 +120,11 @@ def pip_version_check(session, options):
                 process_dependency_links=options.process_dependency_links,
                 session=session,
             )
-            all_candidates = finder.find_all_candidates("pip")
-            if not all_candidates:
+            try:
+                candidate = finder.find_best_candidate("pip", SpecifierSet())
+            except ValueError:
```

Comment:
Since this is a non-trivial function that can cause `ValueError` for potentially lots of different reasons, it seems like you should have a dedicated exception type, if there isn't already one (along the lines of `NoCompatibleCandidateError`, etc).

Reply:
I am not against more fine-grained exception types, but I’m not sure it is desired here. This version check is done automatically after a pip invocation, and it would be very confusing to the user if an uncaught exception occurs here. Maybe it is better to simply catch everything (maybe with a log message asking the user to report) and carry on.

Reply:
It turns out the `CompletedProcess` has an extra benefit of separating the exceptions nicely.